lib/event_log.ex

```diff
@@ -0,0 +1,103 @@
+defmodule EventLog.Consumer do
+  @callback init(Map) :: no_return()
+  @callback handle_event(Map, Map) :: Map
```

Comment:
This is actually `@callback handle_event(%ServerSentEvent{}, state :: %EventLog{}) :: %EventLog{}`

I find this API very wonky, because it exposes the `%EventLog{}` struct to the consumer, who shouldn't have to know about many of the things in `%EventLog{}` struct state.

If we want to keep this approach (with `EventLog` process and a consumer callback module) What I think would be more consistent would be adding a `consumer_state` field to `EventLog` struct and having an interface similar to the following:

```
@type consumer_state :: any()
@callback init(opts :: Keyword.t()) :: {:ok, consumer_state}
@callback handle_event(consumer_state, event_id :: integer, %ServerSentEvent{}) :: {:ok, consumer_state}
```

This would be a simpler interface and wouldn't break the encapsulation as much.

Reply:
(alternatively we can what we were talking about in backend_hq and we can drop the consumer_state altogether and add a third gen_server)